pkg/controller/scaledobject/hpa.go

```diff
@@ -153,10 +157,22 @@ func (r *ReconcileScaledObject) getScaledObjectMetricSpecs(logger logr.Logger, s
 	return scaledObjectMetricSpecs, nil
 }
 
+func getResourceMetrics(resourceMetrics []*autoscalingv2beta2.ResourceMetricSource) []autoscalingv2beta2.MetricSpec {
+	var metrics []autoscalingv2beta2.MetricSpec
```

Comment:
minor nit: I think it's slightly better to preallocate the slice to the correct size since that's known in this case. 

```suggestion
	metrics := make ([]autoscalingv2beta2.MetricSpec, 0, len(resourceMetrics))
```

Reply:
done